docs/sources/developers/plugins/backend/_index.md

```diff
@@ -0,0 +1,62 @@
+
+## Use cases for implementing a backend plugin
+
+The following examples should give you an idea of why you would consider implementing a backend plugin:
+
+- Enable [Grafana Alerting]({{< relref "../../../alerting/rules.md" >}}) for data sources.
+- Connect to non-HTTP services that normally cannot be connected to from a web browser, e.g. SQL database servers.
+- Keep state between users, e.g. query caching for data sources.
+- Use custom authentication methods and/or authorization checks not supported in Grafana.
+- Custom data source request proxy, see [Resources]({{< relref "#resources" >}}).
+
+## Grafana’s backend plugin system
+
+The Grafana backend plugin system is based on the [go-plugin library by HashiCorp](https://github.com/hashicorp/go-plugin). Grafana server launches each backend plugin as a subprocess and communicates with it over [gRPC](https://grpc.io/). This approach has a number of benefits:
+- Plugins can’t crash your grafana process: a panic in a plugin doesn’t panic the server.
+- Plugins are easy to develop: just write a Go application and go build (or use any other language which supports gRPC).
+- Plugins can be relatively secure: The plugin only has access to the interfaces and args given to it, not to the entire memory space of the process.
+
+Grafana's backend plugin system exposes a couple of different capabilities or building blocks that a backend plugin can implement. Currently these are query data, resources, health checks and collect metrics.
+
+### Query data
+
+The query data capability allows a backend plugin to handle data source queries, usually submitted from a [dashboard]({{< relref "../../../features/dashboard/dashboards.md" >}}), [Explore]({{< relref "../../../features/explore/index.md" >}}) or [Grafana Alerting]({{< relref "../../../alerting/rules.md" >}}). The response format contains [data frames]({{< relref "data-frames.md" >}}), which are suitable for visualising metrics, logs, and traces. This capability is required to implement for a backend data source plugin.
+
+### Resources
+
+The resources capability allows a backend plugin to handle custom HTTP requests sent to the Grafana HTTP API and respond with custom HTTP responses. Here, the request and response formats can vary, e.g. JSON, plain text, HTML or static resources (files, images) etc. Compared to the query data capability where the response contains data frames, resources gives the plugin developer a lot of flexibility for extending and open up Grafana for new interesting use cases.
+
+Examples of use cases for implementing resources:
+- Implement a custom data source proxy in case certain authentication/authorization or other requirements is required/needed that's not supported in Grafana's [built-in data proxy](https://grafana.com/docs/grafana/latest/http_api/data_source/#data-source-proxy-calls).
```

Comment:
```suggestion
- Implement a custom data source proxy in case certain authentication/authorization or other requirements are required/needed that are not supported in Grafana's [built-in data proxy](https://grafana.com/docs/grafana/latest/http_api/data_source/#data-source-proxy-calls).
```